src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/RuntimeTypeSystem_1.cs

```diff
@@ -264,7 +264,7 @@ public TypeHandle GetTypeHandle(TargetPointer typeHandlePointer)
 
         if ((addressLowBits != TypeHandleBits.MethodTable) && (addressLowBits != TypeHandleBits.TypeDesc))
         {
-            throw new InvalidOperationException("Invalid type handle pointer");
+            throw new ArgumentException("Invalid type handle pointer", nameof(typeHandlePointer));
```

Comment:
Switched these to match the HRESULT returned by the DAC. Found via @max-charlamb's suggestion in https://github.com/dotnet/runtime/pull/110702#issuecomment-2542425528